src/common.cuh

```diff
@@ -50,7 +50,7 @@ __device__ T *malloc_and_copy(const T *const source, int count)
  * @returns Pointer to the created array
  */
 template<class T>
-__device__ T *device_realloc(T *source, int old_size, int new_size)
+__host__ __device__ T *device_realloc(T *source, int old_size, int new_size)
```

Comment:
`device_realloc` is now `__host__ __device__`, so `device` in the name of this function sounds strange. It relates to #4

Reply:
I think we can ignore it for now and leave it to fixers of #4